### DISCUSSION TITLE
How do you create a new module?

### DISCUSSION THREAD

**Discussion by Orpheus007** [ESchouten/CleanArchitecture#26] [Q&A]
Greeting,
I want to create a new module just like you did in this repo with the adapters, domain and use-cases modules but I don't know how to add the correct type of module with intellij.

**Comment by ESchouten** [comment]
Hi! Thanks for your interest in this repo!
- Add your module in the include block of `settings.gradle.kts`, e.g.
`":adapters:example"`
- In the adapter layer, create the `example` folder
- Within this folder you can add a `build.gradle.kts` where you define your dependencies (see `/adapters/authentication` for an example)
- Also in the `example` folder, create `/src/main/kotlin/example` folders. In this folder you place your code. Gradle will detect it as module automatically, if not, in the gradle menu click `reload all gradle projects` (spinning arrows)

Don't hesitate to ask any other questions if you have any!
Good luck!
